1-Draft/RFCxxxx-Secrets-Management.md

```diff
@@ -0,0 +1,288 @@
+    string name,
+    IReadOnlyDictionary<string, object> parameters,
+    out Exception error);
+
+public abstract bool RemoveSecret(
+    string name,
+    IReadOnlyDictionary<string, object> parameters,
+    out Exception error);
+
+public abstract KeyValuePair<string, string>[] GetSecretInfo(
+    string filter,
+    IReadOnlyDictionary<string, object> parameters,
+    out Exception error);
+```
+
+When PowerShell loads the module, the required assembly will be loaded and the implementing type becomes available.
+
+### PowerShell script implementation
+
+For a script implementation, the PowerShell module must include a subdirectory named 'ImplementingModule' in the same directory containing the module manifest file.
+The ImplementingModule subdirectory must contain PowerShell script module files named 'ImplementingModule' that implements the required script functions.
+
+ImplementingModule.psd1
+
+```powershell
+@{
+    ModuleVersion = '1.0'
+    RootModule = '.\ImplementingModule.psm1'
+    FunctionsToExport = @('Set-Secret','Get-Secret','Remove-Secret','Get-SecretInfo')
+}
+```
+
+ImplementingModule.psm1
+
+```powershell
+function Set-Secret
+{
+    param (
+        [string] $Name,
+        [object] $Secret,
+        [hashtable] $AdditionalParameters
+    )
+}
+
+function Get-Secret
+{
+    param (
+        [string] $Name,
+        [hashtable] $AdditionalParameters
+    )
+}
+
+function Remove-Secret
+{
+    param (
+        [string] $Name,
+        [hashtable] $AdditionalParameters
+    )
+}
+
+function Get-SecretInfo
+{
+    param (
+        [string] $Filter,
+        [hashtable] $AdditionalParameters
+    )
+}
+```
+
+## Vault registration cmdlets
+
+The following cmdlets are provided for vault extension registration.
+
+```powershell
+Register-SecretsVault
+Get-SecretsVault
+Unregister-SecretsVault
+```
+
+`Register-SecretsVault` registers a PowerShell module as an extension vault for the current user context.
+Validation is performed to ensure the module either provides the required binary with implementing type or the required script commands.
+If a dictionary of additional parameters is specified then it will be stored securely in the built-in local vault, assuming that some parameter arguments likely contain secrets.
+
+`Get-SecretsVault` returns a list of extension vaults currently registered in the user context.
+
+`Unregister-SecretsVault` un-registers an extension vault.
+
+`Get-SecretsVault` will enumerate registered extensions returning
+the module name and cmdlet used (if appropriate) and the friendly name.
+
+A `SecretsVaultInfo` object will contain properties for all supported
+parameters by registration.
+
+```output
+Name       Module     Cmdlet               ScriptBlock
+----       ------     ------               -----------
+AzKeyVault AzKeyVault Get-AzKeyVaultSecret
+myVault                                    param($Name)
+```
+
+### Storing Secrets
+
+The `Add-Secret` cmdlet is used to store a secret.

```

Comment:
Everywhere else you've used **Set** as the verb, and then here you switched to **Add** but you're still implementing it as **Set**, where it overwrites by default. If it's going to be named **Add** or **New** it shouldn't clobber by default, it should have a `-Force` parameter rather than a `-NoClobber` switch.